docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,533 @@
+- no need for specifying `spec.targetRef` when targeting `Mesh`
+- easier policy matching code
+
+#### Disadvantages
+
+- can be harder to understand because `spec.targetRef` can be MeshSubset (which is optional and probably will be rarely used)
+- yet another migration for users
+- unable to apply config per routes backendRef 
+
+### Leave Mesh*Routes in top level targetRef as it is now
+
+#### Advantages
+
+- can be simpler as you are always targeting all proxies to which the route applies
+- can target specific backendRef from route
+- no need for additional implementation
+
+#### Disadvantages
+
+- you cannot override route config for subset of proxies without producer/consumer policies (universal use case)
+- mixing proxies and outbounds in `spec.targetRef`
+- when `Mesh*Route` is in `spec.targetRef` targetRef we still have `spec.to[].targetRef` section which can be misleading as route already points to some outbound  
+
+## Design
+
+The design is grounded in multiple recent MADRs around improvements to policies: 
+
+- [MeshService policy matching](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/046-meshservice-policy-matching.md)
+- [Policy on Namespace](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/052-policy-on-namespace.md) - which introduced consumer/producer policies
+- [Policy matching with real resources](https://github.com/kumahq/kuma/pull/10726/files)
+
+### API
+
+New API for policy that selects `Mesh*Route`
+
+```yaml
+type: MeshTimeout
+name: timeouts-on-route
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh | MeshSubset # either Mesh, MeshSubset or MeshGateway
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute | MeshTCPRoute # either MeshHTTPRoute or MeshTCPRoute
+        name: route-to-backend
+      default:
+        http:
+          requestTimeout: 5s
+          streamIdleTimeout: 1h
+```
+
+specifying `Mesh` in `spec.targetRef` will be optional (this probably will be most used case). Minimal policy, targeting
+`MeshHTTPRoute` will look like this:
+
+```yaml
+type: MeshTimeout
+name: timeouts-on-route
+mesh: default
+spec:
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute
+        name: route-to-backend
+      default:
+        http:
+          requestTimeout: 5s
+          streamIdleTimeout: 1h
+```
+
+To introduce this change we need to deprecate specifying `Mesh*Route` in `spec.targetRef`.
+
+Should we allow selecting MeshGateway in `spec.targetRef` when `Mesh*Route` is configured in `spec.to[].targetRef`? probably this 
+is obsolete as Route will select gateway, and we want to apply policy on a route. Is there a need to make it more specific than route? 
+
+### Affected policies
+
+At the moment we can only target `Mesh*Routes` in MeshTimeout policy. This MADR is based on MeshTimeout policy.
+Looking at what can be configured at Envoy route we can also apply this to our MeshRetry policy. Rate limit is also configurable on route,
```

Comment:
There is also MeshLoadBalancingStrategy that configures hashPolicy on routes. Should we add support for MeshHTTPRoute?

Reply:
added section about MeshLoadBalancingStrategy